### ISSUE TITLE
cron try to set non-existant field info

### ISSUE THREAD

**Issue by blink38** [academic-moodle-cooperation/moodle-mod_offlinequiz#27]
Cron.php attempt to set field **info** in table **offlinequiz_queue**, which field does not exist.

` $DB->set_field('offlinequiz_queue', 'info', 'offlinequiz not found', array('id' => $job->id));`

This line make the cron.php script failed and stop :
 
`PHP Notice:  Undefined index: info in /var/www/moodle/lib/dml/mysqli_native_moodle_database.php on line 1457
Default exception handler: Erreur d'écriture vers la base de données Debug: Unknown column 'info' in 'field list'
UPDATE mdl_offlinequiz_queue SET info = ? WHERE id = ?
[array (
  0 => 'offlinequiz not found',
  1 => '10',
)]
Error code: dmlwriteexception
* line 477 of /lib/dml/moodle_database.php: dml_write_exception thrown
* line 1472 of /lib/dml/mysqli_native_moodle_database.php: call to moodle_database->query_end()
* line 1749 of /lib/dml/moodle_database.php: call to mysqli_native_moodle_database->set_field_select()
* line 107 of /mod/offlinequiz/cron.php: call to moodle_database->set_field()
* line 302 of /mod/offlinequiz/cron.php: call to offlinequiz_evaluation_cron()
`

This problem can be reproduce by creating  a quizz, importing response so that they are added to the queue with status "new", and deleting the quizz before the response are computed. Script will enter in first if condition :

 `if (!$offlinequiz = $DB->get_record('offlinequiz', array('id' => $job->offlinequizid)))`

**Comment by Gael-Mifsud** [comment]
Hello,

On Moodle 3.1.1, with this module version 3.1.1, to get round of this problem, we have:
- added a column "info" of type LONGTEXT to the "table offlinequiz_queue_data"
- modified the  type of the column "pagenumber" of "mdl_offlinequiz_scanned_pages" into BIGINT
The cron stops when there is a database error. With these changes, we get rid of theses errors.

Actually, it seems that the problem lies in the page number recognition (we have seen pagenumbers of 2 million and more !).

Naturally, after that, we have to correct the pagnumbers in the activity interface.
But, at least, the cron is  not stopped anymore.

**Comment by univietw** [comment]
@Gael-Mifsud Would you mind making a pull request for this? I think, I understood the problem now:
The Problem is, that the page number is a binary coded field, and it seems like the recognition finds really big pagenumbers, that don't fit into INT.
I will not fix the pagenumber recognition in this state any more, since the code is very complex and I am working on a whole new algorithm for the page scanner, but your fix seems like to be needed in the new algorithm anyway and as it works around this problem, I would be happy to give you the credit for this patch.

**Comment by univietw** [comment]
Hi, I'm so sorry it took me so long to really look into your problem.
At the time you were changing this merge request now I looked deeper into your problem and kind of understood the error and you were right, it has to do with the size of the pagenumbers, as they can be up to 2^25 and this is way too big for the INT-size that the pagenumberfield has.
Concerning the problem, that the queue wants to write into the "info"-field: I have found the insertions of these fields in the upgrade.php already for Version 2013012410
The problem was the missing change in the install.xml which is used for new moodle installations. Seems like your offlinequiz was installed AFTER 2013/24/01 and by this has this problem.
Still, it's interesting, that this problem was never found at our final tests before every release and I will talk about this problem tomorrow in our internal AMC meeting.
I will merge your bug with some small changes to make sure everyone will get this change and the bug will not happen again on new installations.
Thank you very much for your help
